public/js/helpers/chart_helper.js

```diff
@@ -53,3 +95,129 @@ export function padPoints (pts, binSize, sustain) {
   pts.unshift(front)
   pts.push(back)
 }
+
+export function synchronize (dygraphs, syncOptions) {
+  let prevCallbacks = []
+
+  dygraphs.map((g) => {
+    g.ready(() => {
+      var callBackTypes = ['drawCallback', 'highlightCallback', 'unhighlightCallback']
+      for (var j = 0; j < dygraphs.length; j++) {
+        if (!prevCallbacks[j]) {
+          prevCallbacks[j] = {}
+        }
+        for (var k = callBackTypes.length - 1; k >= 0; k--) {
+          prevCallbacks[j][callBackTypes[k]] = dygraphs[j].getFunctionOption(callBackTypes[k])
+        }
+      }
+
+      if (syncOptions.zoom) {
+        attachZoomHandlers(dygraphs, syncOptions, prevCallbacks)
+      }
+
+      if (syncOptions.selection) {
+        attachSelectionHandlers(dygraphs, prevCallbacks)
+      }
```

Comment:
You can move the calls to `attachZoomHandlers` and `attachSelectionHandlers` to after the `dygraphs.map` loop.